api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionService.java

```diff
@@ -115,27 +116,36 @@ public GenomicExtractionService(
         .map(
             dbSubmission -> {
               try {
-                // RW-6537: Don't make a call to Terra for every submission. Submissions in a non
-                // running state will not change
-                WgsCohortExtractionConfig cohortExtractionConfig =
-                    workbenchConfigProvider.get().wgsCohortExtraction;
-                FirecloudSubmission firecloudSubmission =
-                    submissionApiProvider
-                        .get()
-                        .getSubmission(
-                            cohortExtractionConfig.operationalTerraWorkspaceNamespace,
-                            cohortExtractionConfig.operationalTerraWorkspaceName,
-                            dbSubmission.getSubmissionId());
+                // Don't bother checking if we already know the job is in a terminal status.
+                if (dbSubmission.getStatusEnum() == TerraJobStatus.RUNNING
+                    || dbSubmission.getStatusEnum() == null) {
+                  WgsCohortExtractionConfig cohortExtractionConfig =
+                      workbenchConfigProvider.get().wgsCohortExtraction;
+                  FirecloudSubmission firecloudSubmission =
+                      submissionApiProvider
+                          .get()
+                          .getSubmission(
+                              cohortExtractionConfig.operationalTerraWorkspaceNamespace,
+                              cohortExtractionConfig.operationalTerraWorkspaceName,
+                              dbSubmission.getSubmissionId());
+                  dbSubmission.setTerraSubmissionDate(
```

Comment:
is this value available from the response to create submission? If so, we could just cache this value then and not worry about it here.

Reply:
I'll check

Reply:
Unfortunately no - createSubmission just returns a UUID

Reply:
This is my bad for adding all the fields when I implemented it (I'll partially blame Firecloud API docs for not specifying what object is being returned so I had to manually figure it out). But I think we do actually get it from Firecloud but it's not being parsed out right now since it's not part of the API model. 

I invoked createSubmission manually from Terra and got back the following fields.

![image](https://user-images.githubusercontent.com/2770197/117987923-ebac9b00-b308-11eb-8d34-488f18eb8fe0.png)

Reply:
Ooh, thanks for checking that. I'll extend the API then.